sys/kern/mmap.c

```diff
@@ -60,18 +60,29 @@ int do_munmap(vaddr_t addr, size_t length) {
 
   vm_map_t *uspace = proc_self()->p_uspace;
 
+  if (length == 0)
+    return EINVAL;
+
+  /* Round boundaries to contain entire pages from range [addr, addr + length)
+   */
+  vaddr_t right_boundary = roundup(addr + length, PAGESIZE);
+  addr = rounddown(addr, PAGESIZE);
+
   WITH_VM_MAP_LOCK (uspace) {
-    vm_segment_t *seg = vm_map_find_segment(uspace, addr);
-    if (!seg)
-      return EINVAL;
+    while (addr < right_boundary) {
+      vm_segment_t *seg = vm_map_find_segment(uspace, addr);
+      if (!seg)
+        return EINVAL;
+
+      /* TODO We support unmaping entire segments only! */
```

Comment:
Is this comment up to date?

Reply:
You're right. It's not up to date now. I will remove it.